gems/aws-sdk-core/lib/seahorse/client/plugins/endpoint.rb

```diff
@@ -42,7 +40,11 @@ def after_initialize(client)
         class Handler < Client::Handler
 
           def call(context)
-            context.http_request.endpoint = URI.parse(context.config.endpoint.to_s)
+            if !context.config.respond_to?(:regional_endpoint) || # generic
```

Comment:
When would regional endpoint not exist? I think this plugin should continue to blindly set endpoint and logic like this belongs somewhere else.. it should not know or care about endpoints 2.0 logic.

Reply:
regional endpoint doesn't exist in old services.  I think this logic must exist here to fully handle old service cases. 

I guess one option is to actually remove the Searhose Endpoints plugin when endpoints2.0 is used?

Reply:
Check the most recent commit - I've updated our codegen to remove the seahorse endpoints plugin when ep2 is enabled for a service (slightly tricky since its a searhose base default plugin). 

Let me know which of these approaches you prefer.